package/MDAnalysis/core/universe.py

```diff
@@ -992,6 +993,194 @@ def add_Segment(self, **attrs):
         self.segments = SegmentGroup(np.arange(self._topology.n_segments), self)
         # return the new segment
         return self.segments[segidx]
+    
+    def _add_TopologyObjects(self, object_type, values, types=None, guessed=False,
+                           order=None):
+        """Add new TopologyObjects to this Universe
+
+        Parameters
+        ----------
+        object_type : {'bonds', 'angles', 'dihedrals', 'impropers'}
+            The type of TopologyObject to add.
+        values : iterable of tuples, AtomGroups, or TopologyObjects
+            An iterable of: tuples of atom indices, or AtomGroups, 
+            or TopologyObjects. If every value is a TopologyObject, all 
+            keywords are ignored.
+        types : types (optional)
+            Type of bond/angle/dihedral/improper, or atom types
+        guessed : bool (optional)
+            Whether these connections have been guessed.
+        order : int (optional)
+            Bond order
+        """
+        if all(isinstance(x, TopologyObject) for x in values):
+            first = values[0]
+            try:
+                types = first.type
+            except AttributeError:
+                types = None
+            guessed = first.is_guessed
+            order = first.order
+        
+        print(types, guessed, order)
+
+        values = [x.indices if isinstance(x, (AtomGroup, TopologyObject)) 
+                  else x for x in values]
+        
+        try:
+            attr = getattr(self._topology, object_type)
+        except AttributeError:
+            self.add_TopologyAttr(object_type, [])
+            attr = getattr(self._topology, object_type)
+        
+        attr.add_bonds(values, types=types, guessed=guessed, order=order)
+
+    def add_Bonds(self, values, types=None, guessed=False, order=None):
+        """Add new Bonds to this Universe and recreate fragments.
+
+        Parameters
+        ----------
+        values : iterable of tuples, AtomGroups, or Bonds
+            An iterable of: tuples of atom indices, or AtomGroups, 
+            or Bonds. If every value is a Bond, all 
+            keywords are ignored.
```

Comment:
It's not clear from the docstring how the method works with AtomGroups. You need to clarify that one needs to supply a list of AtomGroups with exactly two atoms per group. Maybe you could add an example for creating TIP4P water bonds such as
```python
import MDAnalysis as mda
from MDAnalysisTests.datafiles import GRO
u = mda.Universe(GRO)
ow_hw1 = u.select_atoms("resname SOL and (name OW or name HW1)").split('residue')
ow_hw2 = u.select_atoms("resname SOL and (name OW or name HW2)").split('residue')
ow_mw = u.select_atoms("resname SOL and (name OW or name MW)").split('residue')
u.add_bonds(ow_hw1 + ow_hw2 + ow_mw)
```
(or maybe there's a smarter way to do this)